### DISCUSSION TITLE
Javascript API/SDK?

### DISCUSSION THREAD

**Discussion by ceevee830** [mlflow/mlflow#7784] [Q&A]
Hello, we're looking to use one of your SDKs as an abstraction layer to access the data.  In the docs, we see Python, R, Java, and REST [(here)](https://mlflow.org/docs/latest/index.html).

We also javascript SDK in the repo **mlflow/server/js/src/experiment-tracking/sdk** but not in the docs.

Question; how can we use the Javascript SDK in our app?  
Followup Qs: 

- If it's not available, are there plans to make it available?  
- Where is this package used inside your repo?

**Comment by JHibbard** [comment]
Hi ceevee830,

After looking into the `mlflow/server/js/src/experiment-tracking/sdk` it appears to be generated code. There's also a note that the code generation approach will be changed at some point which may significantly change the generated code. I'd suggest using the REST API docs alongside something like fetch to make a JS sdk. This could also be a good OSS contribution if that's of interest to you, I'm sure there'd be other developers interested in helping you create and maintain this feature.

Hope that helps!
Jim
